sympy/functions/special/hyper.py

```diff
@@ -288,7 +288,58 @@ def _eval_nseries(self, x, n, logx, cdir=0):
             den = Mul(*[RisingFactorial(b, i) for b in bq])
             terms.append(((num/den) * (arg**i)) / factorial(i))
 
-        return (Add(*terms) + Order(x**n,x))
+        result = Add(*terms)
+        newn = min((Abs(a) for a in ap if a.is_negative and a.is_integer), default=n)
+
+        if newn < n:
+            return result
+        return result + Order(x**n, x)
+
+    def _eval_aseries(self, n, args0, x, logx):
+
+        # The asymptotic expansion of the hypergeometric functions is quite complicated
+        # and are not well defined for all relation between the parameters p and q.
+        # The following is a simple implementation of the asymptotic expansion of the
+        # hypergeometric function for the case when p = q + 1 and |z| -> oo.
+
+        from sympy.functions.special.gamma_functions import gamma
+        from sympy.series.order import Order
+
+        ap = self.args[0]
+        bq = self.args[1]
+        z = unpolarify(self.args[2])
+
+        if len(ap) != len(bq) + 1:
+            return super()._eval_aseries(n, args0, x, logx)
+
+        for i in range(len(ap)):
+            for l in range(i+1, len(ap)):
+                if (ap[i]-ap[l]).is_integer:
+                    return super()._eval_aseries(n, args0, x, logx)
```

Comment:
This should be `if (ap[i]-ap[l]).is_integer is not False:`

https://docs.sympy.org/latest/guides/booleans.html